lib/server.js

```diff
@@ -121,17 +121,35 @@ WalletService.prototype.createWallet = function(opts, cb) {
     return cb(new ClientError('Invalid public key'));
   };
 
-  var wallet = Wallet.create({
-    name: opts.name,
-    m: opts.m,
-    n: opts.n,
-    network: network,
-    pubKey: pubKey.toString(),
-  });
+  var newWallet;
+  async.series([
+
+    function(acb) {
+      if (!opts.id)
+        return acb();
 
-  self.storage.storeWallet(wallet, function(err) {
-    log.debug('Wallet created', wallet.id, network);
-    return cb(err, wallet.id);
+      self.storage.fetchWallet(opts.id, function(err, wallet) {
+        if (wallet) return acb(new ClientError('Wallet already exists'));
```

Comment:
Maybe return an error code like `WEXISTS` so the clien can more reliably deal with this error?